docs/fundamentals/code-analysis/quality-rules/il3002.md

```diff
@@ -0,0 +1,24 @@
+---
+title: "IL3002: Avoid calling members annotated with 'RequiresAssemblyFilesAttribute' when publishing as a single file (code analysis)"
+description: "Members annotated with 'RequiresAssemblyFilesAttribute' require assembly files to be on disk"
+ms.date: 04/15/2021
+ms.topic: reference
+author: mateoatr
+ms.author: matorre
+f1_keywords:
+  - "IL3002"
+  - "RequiresAssemblyFiles"
+---
+# IL3002: Avoid calling members annotated with 'RequiresAssemblyFilesAttribute' when publishing as a single file.
+
+## Cause
+
+When publishing as a single-file (for example, by setting the PublishSingleFile property in a project to true), calling members annotated with the `RequiresAssemblyFilesAttribute` attribute might be unsafe, given that members annotated with this attribute require assembly files to be on disk and the assemblies embedded in a single-file are memory loaded.
```

Comment:
```suggestion
When publishing an app as a single file (for example, by setting the `PublishSingleFile` property in a project to `true`), calling members annotated with the `RequiresAssemblyFilesAttribute` attribute might be unsafe. These calls can be unsafe because members annotated with this attribute require assembly files to be on disk, and the assemblies embedded in a single-file app are memory loaded.
```